### DISCUSSION TITLE
Cookies

### DISCUSSION THREAD

**Discussion by ipftalabua** [sqlpage/SQLPage#101] [Q&A] [answered]
I was just trying the cookie example as documented on

https://sql.ophir.dev/documentation.sql?component=cookie#component

However, sqlpage tells me

`Couldn't load "sqlpage/templates\\cookie.handlebars" into cache`

So I checked whether my version of sqlpage was out of date by looking at the Github template folder, not finding `cookie.handlebars`, either. Am I missing something?

Thanks in advance!

**Comment by lovasoa** [comment, marked as answer]
Hi ! 

The error message is terrible, we should work on that. What it really means is that you are trying to set a cookie after having already rendered elements on the page, whereas the cookie component should always be used at the very top of the page.

> Since cookies need to be set before the response body is sent to the client, this component should be placed at the top of the page, before any other components that generate output.
>
> *https://sql.ophir.dev/documentation.sql?component=cookie#component*

**Comment by ipftalabua** [comment]
Thanks, this is what I was missing.
